qiime/core/archiver.py

```diff
@@ -212,6 +218,20 @@ def _formatted_provenance(self):
                 'parameter-references': self.provenance.parameter_references
             }
 
+    def get_index_paths(self):
+        # TODO: this extracts the entire archive, even though that isn't
+        # necessary to get the index paths. Refactor to avoid this.
+        if not os.path.exists(self._data_dir):
+            self._extract_data()
+        result = {}
+        for path in os.listdir(self._data_dir):
+            if os.path.isfile(os.path.join(self._data_dir, path)):
+                filename = os.path.split(path)[1]
+                if filename.startswith('index.'):
+                    ext = os.path.splitext(filename)[1]
+                    result[ext[1:]] = os.path.join(self._data_dirname, path)
```

Comment:
There could be multiple files with the same extension, e.g. `foo.bar.png` and `foo.baz.png`. Can you store a `set` of files with a given extension?